# Preparing Dataset 

In [3]:
import os
import cv2
import pandas as pd

# Rename the videos in sequential numbering

In [14]:
#path = 'D:/Workspace/__Workstation/4th year project/CrashCatcher-master/My work/Positive' ### specify the path
#files = os.listdir(path)
#i = 1

#for file in files:
    #os.rename(os.path.join(path, file), os.path.join(path, file[:-10]+'%03d' % i+'.mp4'))
    #i = i+1
    

# Extracting meta data and exporting as CSV
    

In [6]:
download_dir = "Meta_Data_Positive.csv" #where you want the file to be downloaded to 
csv = open(download_dir, "w") 
#"w" indicates that you're writing strings to the file
columnTitleRow = "Video, frames, width, height, fps\n"
csv.write(columnTitleRow)

path = 'D:/Workspace/__Workstation/4th year project/CrashCatcher-master/My work/Positive/' ### specify the path
files = os.listdir(path)

for file in files:

    cap = cv2.VideoCapture(path+file)

    if not (cap.isOpened()): 
        print ("could not open :"+file)
        
    #frames = str(cap.get(7))
    
    frames = str(cap.get(cv2.CAP_PROP_FRAME_COUNT))
    width  = str(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
    height = str(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
    fps    = str(cap.get(cv2.CAP_PROP_FPS))

    row = file + "," + frames +"," + width +"," + height + "," + fps + "\n"
    print (row)
    #uncomment the below line
    csv.write(row)

vehicle_guardrail_001.mp4,90.0,1280.0,720.0,29.97

vehicle_guardrail_002.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_003.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_004.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_005.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_006.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_007.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_008.mp4,90.0,1280.0,720.0,29.97

vehicle_guardrail_009.mp4,90.0,1280.0,720.0,29.97

vehicle_guardrail_010.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_011.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_012.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_013.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_014.mp4,90.0,1280.0,720.0,29.97

vehicle_guardrail_015.mp4,90.0,1280.0,720.0,29.97

vehicle_guardrail_016.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_017.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_018.mp4,90.0,1280.0,720.0,29.97

vehicle_guardrail_019.mp4,120.0,1280.0,720.0,29.97

vehicle_guardrail_

# Chopping extra frames to make Max frames as 120 or less

In [ ]:
from moviepy.editor import VideoFileClip


long=['NoAccident_014.mp4']

for x in long:
        clip = VideoFileClip(x) 
        newclip = clip.subclip(0,4)
        ### CAUTION this function gets 4 sec video only if it longer than 4 .. running it multiple times may give you just frames
        newclip.write_videofile(x,audio=False) # don't render audio.  
        print(x)

